dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerImageApi.java

```diff
@@ -0,0 +1,162 @@
+import org.jclouds.collect.IterableWithMarker;
+import org.jclouds.collect.PagedIterable;
+import org.jclouds.collect.internal.ArgsToPagedIterable;
+import org.jclouds.dimensiondata.cloudcontrol.DimensionDataCloudControlApi;
+import org.jclouds.dimensiondata.cloudcontrol.domain.CustomerImage;
+import org.jclouds.dimensiondata.cloudcontrol.domain.CustomerImages;
+import org.jclouds.dimensiondata.cloudcontrol.domain.OsImage;
+import org.jclouds.dimensiondata.cloudcontrol.domain.OsImages;
+import org.jclouds.dimensiondata.cloudcontrol.domain.PaginatedCollection;
+import org.jclouds.dimensiondata.cloudcontrol.filters.DatacenterIdFilter;
+import org.jclouds.dimensiondata.cloudcontrol.filters.OrganisationIdFilter;
+import org.jclouds.dimensiondata.cloudcontrol.options.PaginationOptions;
+import org.jclouds.http.filters.BasicAuthentication;
+import org.jclouds.http.functions.ParseJson;
+import org.jclouds.json.Json;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.Transform;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+@RequestFilters({ BasicAuthentication.class, OrganisationIdFilter.class })
+@Consumes(MediaType.APPLICATION_JSON)
+@Path("/{jclouds.api-version}/image")
+public interface ServerImageApi {
+
+   @Named("image:list")
+   @GET
+   @Path("/osImage")
+   @ResponseParser(ParseOsImages.class)
+   @RequestFilters(DatacenterIdFilter.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   PaginatedCollection<OsImage> listOsImages(PaginationOptions options);
+
+   @Named("image:list")
+   @GET
+   @Path("/osImage")
+   @Transform(ParseOsImages.ToPagedIterable.class)
+   @ResponseParser(ParseOsImages.class)
+   @RequestFilters(DatacenterIdFilter.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<OsImage> listOsImages();
+
+   @Named("image:list")
+   @GET
+   @Path("/customerImage")
+   @ResponseParser(ParseCustomerImages.class)
+   @RequestFilters(DatacenterIdFilter.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   PaginatedCollection<CustomerImage> listCustomerImages(PaginationOptions options);
+
+   @Named("image:list")
+   @GET
+   @Path("/customerImage")
+   @Transform(ParseCustomerImages.ToPagedIterable.class)
+   @ResponseParser(ParseCustomerImages.class)
+   @RequestFilters(DatacenterIdFilter.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<CustomerImage> listCustomerImages();
+
+   @Named("image:get")
+   @GET
+   @Path("/osImage/{id}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   OsImage getOsImage(@PathParam("id") String id);
+
+   @Named("image:get")
```

Comment:
[minor] do we want to apply same name as line 90 as this method will get the same config as the above?